### PR TITLE
Remove all references to TAP and add doc on how to manage users/gateways

### LINKED ISSUE #156: Document how to use the Cloud Portal (Admin)
We have *some* information on how to use the Cloud Portal but some aspects are not yet documented (or not well enough). We should add some more information on those topics (but not necessarily limited to):
* How to add a new user / edit existing users
* How to add a new gateway / edit an existing gateway
* How to tie a user to a gateway

### PR DESCRIPTION
**Pull request by ichbinblau**
1. remove all the code logics and settings for TAP; 
2. remove test_harness; 
3. add document how a user is mapped to a gateway and close #156
   - created a [wiki ](https://github.com/01org/SmartHome-Demo/wiki/How-to-manage-users-and-gateways-in-the-admin-portal) on how to manage users and gateways
   - add bullets 6 and 7 in `tools/docker/README.md` to reference the wikis
4. Solve the celery warning "The AMQP result backend is scheduled for deprecation in version 4.0 and removal in version v5.0.     Please use RPC backend or a persistent backend." on boot by updating the protocol
5. Create symbolic link to admin's portal model image
   Since the model images are generated at run-time in the admin portal, a soft link is created to point to the admin portal's model image folder. It was not a problem in TAP or Cloud foundry cos they access the images via the `http://<hostname>:4000/images/model` directly. In docker env, there is no way to get host's hostname from the container.

The updates has been self tested and the image can be downloaded by `docker pull xshan1/smarthome-demo:156`

Signed-off-by: Theresa <theresa.shan@intel.com>

### CODE REVIEW

**Comment by gvancuts** [comment]
:+1: Looks good to me @ichbinblau !

Could someone else do a sanity check on this independantly?

**Comment by uhofemei** [comment]
Done. Only a few minor issues I could find.

**Comment by gvancuts** [comment]
@uhofemei what are those 'minor issues'?

I can't make out if your comment is a thumbs up or thumbs down.

**Comment by uhofemei** [comment]
I was commenting inside the commit on a by line basis. Let me check.

**Comment by uhofemei** [comment]
I see what happened. I commented on the wrong PR from my phone. The comment I made is about https://github.com/01org/SmartHome-Demo/pull/160, which needs some minor tweaking.

**Comment by gvancuts** [comment]
OK, have you tested this one at all then?

**Comment by uhofemei** [comment]
Not yet.

File: smarthome-web-portal/CeleryTask/celeryconfig.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 mq_conn_str = config.get_rabbitmq_conn_str()
 CELERY_TASK_SERIALIZER = 'json'
 CELERY_ACCEPT_CONTENT = ['json']
-CELERY_RESULT_BACKEND = 'amqp://'
+CELERY_RESULT_BACKEND = 'rpc://'
```
Comment (gvancuts) [inline smarthome-web-portal/CeleryTask/celeryconfig.py]: Hi @ichbinblau , is it expected that 'amqp' still is a dependency of the Cloud Portal (and listed in requirements.txt). I did a grep through the source code and saw that 'amqp' is used [here](https://github.com/ichbinblau/SmartHome-Demo/blob/b162546659774bd4f1f27ae59dc214e588409720/smarthome-web-portal/utils/config.py#L94).

Comment (ichbinblau) [inline smarthome-web-portal/CeleryTask/celeryconfig.py]: Yes, it is still required. Cos amqp lib is a core protocol used to interact with Rabbitmq server. We still use Rabbitmq as the Broker to store tasks (Plz check the config [here](https://github.com/ichbinblau/SmartHome-Demo/blob/b162546659774bd4f1f27ae59dc214e588409720/smarthome-web-portal/CeleryTask/celeryconfig.py#L12 ) )

**Comment by uhofemei** [comment]
The file hasn't been modified as part of this PR, but I think we should remove the TAP reference here. https://github.com/01org/SmartHome-Demo/blame/master/smarthome-web-portal/README.md#L4

**Comment by uhofemei** [comment]
LGTM 👍

**Comment by ichbinblau** [comment]
@uhofemei,  are we supposed to replace it with Cloud Foundry or Kubernetes later?

**Comment by gvancuts** [comment]
> @uhofemei, are we supposed to replace it with Cloud Foundry or Kubernetes later?

Yes, can you update the text accordingly? We do not have clear instructions for those yet but this is one of the next item we'll tackle.

**Comment by ichbinblau** [comment]
Hi, @gvancuts. I have updated it to Cloud Foundry for now.

**Comment by uhofemei** [comment]
LGTM 👍

**Comment by gvancuts** [comment]
We got a 👍 from a couple of folks and no other comments... merging this!

**Pull request merged**